index.js

```diff
@@ -44,5 +35,28 @@ function clipboardCopy (text) {
 
   return success
     ? Promise.resolve()
-    : Promise.reject(new DOMException('The request is not allowed', 'NotAllowedError'))
+    : Promise.reject(err)
+}
+
+function copyViaClipboardApi (text) {
+  // Use the Async Clipboard API when available. Requires a secure browsing
+  // context (i.e. HTTPS)
+  return navigator.clipboard
+    ? navigator.clipboard.writeText(text)
+    : Promise.reject() // eslint-disable-line prefer-promise-reject-errors
+}
+
+function clipboardCopy (text) {
+  var clipboardApiError
+  // Use the Async Clipboard API when available. Requires a secure browsing
+  // context (i.e. HTTPS)
+  return copyViaClipboardApi(text)
+    .catch(function (err) {
+      // fallback to document.execCommand() if it Clipboard API or not available
+      clipboardApiError = err
+      return copyViaExecCommand(text)
+    })
+    .catch(function (err) {
+      throw err || clipboardApiError || new DOMException('The request is not allowed', 'NotAllowedError')
```

Comment:
here we reject with error thrown by `document.execCommand()` if any,
with error thrown by Clipboard API with any or fallback NotAllowedError